pypeit/ginga.py

```diff
@@ -54,60 +51,88 @@ def connect_to_ginga(host='localhost', port=9000, raise_err=False):
     return viewer
 
 
-def show_image(inp, chname='Image', waveimg=None, bitmask = None, exten = 0, cuts = None, clear=False, wcs_match = False):
-    """ Displays input image in Ginga viewer
-    Supersedes method in xastropy
-
-    Parameters
-    ----------
-    inp : str or ndarray (2D)
-      If str, assumes the image is written to disk
-
-    Optional Parameters
-    ----------
-    waveimg : str, optional
-      If included, use this in WCS.  Mainly to show wavelength array
-
-    bitmask: ndarray (2D)
-      bitmask produced by PypeIt extraction illustrating which pixels were masked and why
-
-    exten: int, optional
-      extension of image in fits file. Only passed in if inp is a file
-
-    Returns
-    -------
-
+def show_image(inp, chname='Image', waveimg=None, bitmask=None, mask=None, exten=0, cuts=None,
```

Comment:
I don't like that the user has to supply this bitmask object as an argument into the call. There is no reason for this extra argument, why not just directly import bitmask. These show_image routines are at the moment only every going to use a single bitmask, which is that created by the scienceimage class. If we decided to add additional bit masks in the future then we can make this change. 

Also, have you checked all the dependencies on show_image to see that they are not now going to break because of this change? See for example the show method in science image, or the show_twod.py script. 

We should create tests for these. But I would try to grep show_image for the moment, because it is used by many routines.  

Joe

Reply:
Good point about the other scripts.  I'll fix this.

About passing bitmask and mask, you can't import scienceimage into ginga and ginga into scienceimage.  It creates a circular import.  You solved this by putting the import into the function but this is bad coding practice.  Imports go at the top.  So you can't only pass the mask data without also passing the bitmask object that interprets it.  There are a few ways around this, if we're really concerned about minimizing the number of arguments.

Reply:
I guess we can make the bitmask class a separate module to solve this?